pkg/api/dashboard.go

```diff
@@ -99,6 +99,16 @@ func (hs *HTTPServer) GetDashboard(c *contextmodel.ReqContext) response.Response
 		return rsp
 	}
 
+	// V2 values should be read from the k8s API
+	if strings.HasPrefix(dash.APIVersion, "v2") {
+		root := hs.Cfg.AppSubURL
+		if !strings.HasSuffix(root, "/") {
+			root += "/"
+		}
+		url := fmt.Sprintf("%sapis/dashboard.grafana.app/%s/namespaces/%s/dashboards/%s", root, dash.APIVersion, hs.namespacer(c.OrgID), dash.UID)
+		return response.Redirect(url)
```

Comment:
Our options here are:
1. return the payload normally -- it won't match anything else in the legacy API, but will have the same wrapper metadata
2. return an error
3. redirect to the new API

I think 2 or 3 is more appropriate than 1.  3 (implemented here) may be confusing because many clients will automatically redirect

Reply:
@ryantxu my understanding is that converting v2 -> v1 would result in data being lost, but would the lost data be an issue for v1 clients or only clients that need v2 features?